File: Application/app/src/main/java/com/example/sage/foodsharebook/models/Dish.java
package com.example.sage.foodsharebook.models;

import java.io.Serializable;
import java.util.List;
import com.google.gson.annotations.SerializedName;


public class Dish implements Serializable {

	@SerializedName("dish_ingredients")
	private List<DishIngredient> dishIngredients;

	@SerializedName("image")
	private String image;

	@SerializedName("user_id")
	private Integer userId;

	@SerializedName("name")
	private String name;

	@SerializedName("recipe")
	private String recipe;

	@SerializedName("description")
	private String description;

	@SerializedName("id")
	private Integer id;

	public Dish(String name, String recipe, String description, String image,  List<DishIngredient> dishIngredients){
		this.name = name;
		this.recipe = recipe;
		this.description = description;
		this.dishIngredients = dishIngredients;
		this.image = image;
	}

	public void setDishIngredients(List<DishIngredient> dishIngredients){
		this.dishIngredients = dishIngredients;
	}


	public List<DishIngredient> getDishIngredients(){
		return dishIngredients;
	}

	public void setImage(String image){
		this.image = image;
	}

	public String getImage(){
		return image;
	}

	public void setUserId(int userId){
		this.userId = userId;
	}

	public Object getUserId(){
		return userId;
	}

	public void setName(String name){
		this.name = name;
	}

	public String getName(){
		return name;
	}

	public void setRecipe(String recipe){
		this.recipe = recipe;
	}

	public String getRecipe(){
		return recipe;
	}

	public void setDescription(String description){
		this.description = description;
	}

	public String getDescription(){
		return description;
	}

	public void setId(int id){
		this.id = id;
	}

	public int getId(){
		return id;
	}

	@Override
	public String toString(){
		return
				"Response{" +
						",id = '" + id + '\'' +
						"dish_ingredients = '" + dishIngredients + '\'' +
						",image = '" + image + '\'' +
						",user_id = '" + userId + '\'' +
						",name = '" + name + '\'' +
						",recipe = '" + recipe + '\'' +
						",description = '" + description + '\'' +
						"}";
	}
}

File: Application/app/src/main/java/com/example/sage/foodsharebook/models/Dish.kt
package com.example.sage.foodsharebook.models

import java.io.Serializable
import com.google.gson.annotations.SerializedName


class Dish(@field:SerializedName("name")
           var name: String?, @field:SerializedName("recipe")
           var recipe: String?, @field:SerializedName("description")
           var description: String?, @field:SerializedName("image")
           var image: String?, @field:SerializedName("dish_ingredients")
           var dishIngredients: List<DishIngredient>?) : Serializable {

    @SerializedName("user_id")
    private var userId: Int? = null

    @SerializedName("id")
    private var id: Int? = null

    fun setUserId(userId: Int) {
        this.userId = userId
    }

    fun getUserId(): Any? {
        return userId
    }

    fun setId(id: Int) {
        this.id = id
    }

    fun getId(): Int {
        return id!!
    }

    override fun toString(): String {
        return "Response{" +
                ",id = '" + id + '\''.toString() +
                "dish_ingredients = '" + dishIngredients + '\''.toString() +
                ",image = '" + image + '\''.toString() +
                ",user_id = '" + userId + '\''.toString() +
                ",name = '" + name + '\''.toString() +
                ",recipe = '" + recipe + '\''.toString() +
                ",description = '" + description + '\''.toString() +
                "}"
    }
}